### ISSUE TITLE
Feature request: Forward -W flags to clang

### ISSUE THREAD

**Issue by practicalswift** [NASA-SW-VnV/ikos#66]
```
$ cat testcase.cpp
int main() {
  int a{2147483648};
}
$ clang++-7 testcase.cpp && echo "compilation successful."
testcase.cpp:2:9: error: constant expression evaluates to 2147483648 which cannot be narrowed to type 'int' [-Wc++11-narrowing]
  int a{2147483648};
        ^~~~~~~~~~
…
1 warning and 1 error generated.
$ clang++-7 -Wno-c++11-narrowing testcase.cpp && echo "compilation successful."
…
compilation successful.
$ ikos -Wno-c++11-narrowing testcase.cpp && echo "compilation successful."
usage: ikos [options] file[.c|.cpp|.bc]
ikos: error: unrecognized arguments: -Wno-c++11-narrowing
$ ikos testcase.cpp && echo "compilation successful."
[*] Compiling testcase.cpp
testcase.cpp:2:9: error: constant expression evaluates to 2147483648 which cannot be narrowed to type 'int' [-Wc++11-narrowing]
  int a{2147483648};
        ^~~~~~~~~~
…
ikos: error while compiling testcase.cpp, abort.
```

Note that the compilation succeeds with `-Wno-c+11-narrowing`.

**Comment by arthaud** [comment]
Hi @practicalswift,

Good idea.

The current version compiles with `-Wall`, which is a good idea since it can catch a few bugs that ikos cannot catch (see your own example above).

If the user provides `-Wsomething`, should we prepend/append automatically `-Wall` or not? What do you think?

**Comment by practicalswift** [comment]
The default `-Wall` makes perfect sense.

And if the user opts in explicitly to `-Wsomething` I think we still should prepend the default (`-Wall`). The user is free to disable unwanted default warnings explicitly using `-Wno-some-default-warning`.

Speaking from my own experience that is my expectation: that any explicit `-Wsomething` is incremental to what would have been there without `-Wsomething` :-)

**Comment by arthaud** [comment]
Agreed. This is implemented in 4bc57aed5cf7e982eef7d0cb0148f53c3547dac2.
